src/main/java/cubicchunks/asm/CubicChunksMixinConfig.java

```diff
@@ -43,18 +41,44 @@
 import com.google.gson.stream.JsonReader;
 import com.google.gson.stream.JsonWriter;
 
+import it.unimi.dsi.fastutil.objects.Object2BooleanLinkedOpenHashMap;
+import it.unimi.dsi.fastutil.objects.Object2BooleanMap;
+import net.minecraftforge.fml.relauncher.FMLInjectionData;
+
 /**
  * This Mixin configuration plugin class launched from cubicchunks.mixin.selectable.json.
  * Note, that this plugin is not working in JUnit tests due to missing field of 
  * FMLInjectionData required for MinecraftForge configuration used here.
  * Therefore two Mixin classes with an injection in a same method and with a same priority will cause Mixin to fail. */
 public class CubicChunksMixinConfig implements IMixinConfigPlugin {
 
+    private final Object2BooleanMap<String> modDependencyConditions = new Object2BooleanLinkedOpenHashMap<String>();
+    
     @Nonnull
     public static Logger LOGGER = LogManager.getLogger("CubicChunksMixinConfig");
 
     @Override
     public void onLoad(String mixinPackage) {
+        boolean isOptifineCoreModLoaded = true;
+        try {
+            Class.forName("optifine.OptiFineForgeTweaker");
```

Comment:
They use: https://github.com/MinecraftForge/MinecraftForge/blob/9c7538db8de0e385f816c4be37b829673a49a5d9/src/main/java/net/minecraftforge/fml/client/FMLClientHandler.java#L315
I have no idea why this is working.

Reply:
I think it isn't working and that's why you don't see optifine in mod list.

Reply:
Turns out it does work because the code won't throw exception if the file isn't there. And see that https://github.com/MinecraftForge/MinecraftForge:blob/9c7538db8de0e385f816c4be37b829673a49a5d9/src/main/java/net/minecraftforge/fml/client/FMLClientHandler.java#L525-L528

Reply:
I can use that, but I need somehow to detect side (Client or Server) on this stage, because FMLClientHandler.java obviously will not be present in dedicated server.

Reply:
Nope, I can't use it. `FMLClientHandler.instance().hasOptifine()` always return `true`.

Reply:
hm... maybe OptiFine isn't loaded yet at that point or not detected yet.